### PR TITLE
[chore] - front: connection edition flow

### LINKED ISSUE #6695: [Groups UI] - Edition Flow

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR introduces a new DataSourceEditionModal component to improve the data source connection management flow. It allows users to view and edit connection details, including permissions and ownership. 

Key changes include:
- New DataSourceEditionModal component for managing data source connections
- Integration of the modal in the data sources management page
- Support for requesting access to data sources from non-owners

This change aims to simplify the process of managing data source connections and improve the overall user experience for both administrators and regular users.

**References:**
- [Figma](https://www.figma.com/design/QOxHwppG64GtPocpDhS6Y7/Product?node-id=8487-51614&t=qA6a8uWSz9YaYtry-0)
- https://github.com/dust-tt/dust/issues/6695

## Risk

Important risk of breaking existing UI interactions on the Connection Management screen.

## Deploy Plan

- Deploy to `front-edge`
- Deploy to `front`

### CODE REVIEW

File: front/components/data_source/DataSourceEdition.tsx [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+import { InformationCircleIcon } from "@heroicons/react/20/solid";
+import type { NextRouter } from "next/router";
+import React, { useContext, useState } from "react";
+
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { CONNECTOR_CONFIGURATIONS } from "@app/lib/connector_providers";
+import { formatTimestampToFriendlyDate } from "@app/lib/utils";
+import { setupConnection } from "@app/pages/w/[wId]/builder/data-sources/managed";
+
+export type DataSourceIntegration = {
```
Comment (flvndvd) [inline front/components/data_source/DataSourceEdition.tsx:30]: IIRC, this type already exist in `managed.tsx`, you introduced it a few days, hours ago?

File: front/components/data_source/DataSourceEdition.tsx [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+  connectorProvider: ConnectorProvider | null | undefined;
+  dataSourceIntegration: DataSourceIntegration | null;
+  onClose: () => void;
+  router: NextRouter;
+  dustClientFacingUrl: string;
+  user: UserType;
+  setIsRequestDataSourceModalOpen: (show: boolean) => void;
+}
+
+export function DataSourceEditionModal({
```
Comment (flvndvd) [inline front/components/data_source/DataSourceEdition.tsx:60]: What do we do with `DataSourceDetailsModal` which is the current implementation of this (in a simpler version)?

File: front/components/data_source/DataSourceEdition.tsx (hunk 1 above)
Comment (JulesBelveze) [reply front/components/data_source/DataSourceEdition.tsx:30]: I messed up my rebase, it's supposed to be here now.

File: front/components/data_source/DataSourceEdition.tsx (hunk 2 above)
Comment (JulesBelveze) [reply front/components/data_source/DataSourceEdition.tsx:60]: Looking at the Figma, is seems to still exist.. IMHO the only way to edit the permissions should be through this `DataSourceEditionModal` and not when selecting data

File: front/components/data_source/DataSourceEdition.tsx [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { CONNECTOR_CONFIGURATIONS } from "@app/lib/connector_providers";
+import { formatTimestampToFriendlyDate } from "@app/lib/utils";
+import type { PostManagedDataSourceRequestBody } from "@app/pages/api/w/[wId]/data_sources/managed";
+import { setupConnection } from "@app/pages/w/[wId]/builder/data-sources/managed";
+
+interface DataSourceEditionModalProps {
+  isOpen: boolean;
+  owner: WorkspaceType;
+  connectorProvider: ConnectorProvider | null | undefined;
```
Comment (flvndvd) [inline front/components/data_source/DataSourceEdition.tsx:35]: Can we stick to `ConnectorProvider | null`?

File: front/components/data_source/DataSourceEdition.tsx [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+  dataSourceIntegration: DataSourceIntegration | DataSourceType | null;
+  onClose: () => void;
+  router: NextRouter;
```
Comment (flvndvd) [inline front/components/data_source/DataSourceEdition.tsx:38]: We usually avoid passing router and prefer using `useRouter`

File: front/components/data_source/DataSourceEdition.tsx [hunk 5]
```diff
@@ -0,0 +1,436 @@
...
+    integrations: (prev: DataSourceIntegration[]) => DataSourceIntegration[]
+  ) => void;
+  setIsLoadingByProvider?: (
+    providers: (
+      prev: Record<ConnectorProvider, boolean | undefined>
+    ) => Record<ConnectorProvider, boolean | undefined>
+  ) => void;
+}
+
+const REDIRECT_TO_EDIT_PERMISSIONS = [
```
Comment (flvndvd) [inline front/components/data_source/DataSourceEdition.tsx:52]: Can we use proper typing here so the linter will fail if we add a new connector. We can probably do `Omit<..., ...>`

File: front/components/data_source/DataSourceEdition.tsx [hunk 6]
```diff
@@ -0,0 +1,436 @@
...
+import { CONNECTOR_CONFIGURATIONS } from "@app/lib/connector_providers";
+import { formatTimestampToFriendlyDate } from "@app/lib/utils";
+import type { PostManagedDataSourceRequestBody } from "@app/pages/api/w/[wId]/data_sources/managed";
+import { setupConnection } from "@app/pages/w/[wId]/builder/data-sources/managed";
+
+interface DataSourceEditionModalProps {
+  isOpen: boolean;
+  owner: WorkspaceType;
+  connectorProvider: ConnectorProvider | null | undefined;
+  dataSourceIntegration: DataSourceIntegration | DataSourceType | null;
```
Comment (flvndvd) [inline front/components/data_source/DataSourceEdition.tsx:36]: I'm not sure to see which call site passes a `DataSourceIntegration` but I'll try to ensure that this component only accept `DataSourceType`.

File: front/components/data_source/DataSourceEdition.tsx [hunk 7]
```diff
@@ -0,0 +1,436 @@
...
+  let dataSourceName: string | null = null;
+  if (isDataSourceIntegration(dataSourceIntegration)) {
+    isSetup = !!dataSourceIntegration?.connector;
+    dataSourceName = dataSourceIntegration.dataSourceName;
+  } else {
+    isSetup = true;
+    dataSourceName = dataSourceIntegration.name;
+  }
+
+  let dataSourceOwner: EditedByUser | null | undefined = null;
```
Comment (flvndvd) [inline front/components/data_source/DataSourceEdition.tsx:104]: Can we only use one nullish type?

File: front/components/data_source/DataSourceEdition.tsx [hunk 8]
```diff
@@ -0,0 +1,436 @@
...
+          : `/api/w/${owner.sId}/data_sources/managed`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            provider,
+            connectionId: connectionIdRes.value,
+            name: undefined,
```
Comment (flvndvd) [inline front/components/data_source/DataSourceEdition.tsx:146]: It feels weird to pass undefined values here

Comment (flvndvd) [reply front/components/data_source/DataSourceEdition.tsx:146]: Only providing `connectionId` should be enough IIRC.

File: front/components/data_source/DataSourceEdition.tsx [hunk 9]
```diff
@@ -0,0 +1,436 @@
...
+    );
+    if (updateRes.error) {
+      sendNotification({
+        type: "error",
+        title: "Failed to update the permissions of the Data Source",
+        description: updateRes.error,
+      });
+    }
+  };
+  return (
```
Comment (flvndvd) [inline front/components/data_source/DataSourceEdition.tsx:263]: I feel like most of this code is not used, at least I can't see it when browsing the app locally. Is it some legacy behavior where you thought that this was the entry point to setup new connections?

File: front/components/data_source/DataSourceEdition.tsx (hunk 8 above)
Comment (JulesBelveze) [reply front/components/data_source/DataSourceEdition.tsx:146]: Looking into `PostManagedDataSourceRequestBodySchema` you actually need to pass `configuration` and `name`

File: front/components/data_source/DataSourceEdition.tsx (hunk 5 above)
Comment (JulesBelveze) [reply front/components/data_source/DataSourceEdition.tsx:52]: I ended up using `Extract`, seemed convenient in this case 😇

File: front/components/data_source/DataSourceEdition.tsx (hunk 6 above)
Comment (JulesBelveze) [reply front/components/data_source/DataSourceEdition.tsx:36]: Now that we only display this modal within the `data-sources/[name]` view, we can actually remove it yes 👍🏼

File: front/components/data_source/DataSourceEdition.tsx (hunk 9 above)
Comment (JulesBelveze) [reply front/components/data_source/DataSourceEdition.tsx:275]: Well this code is triggered when clicking the "Edit Permissions" button ?

File: front/components/data_source/DataSourceEdition.tsx (hunk 6 above)
Comment (JulesBelveze) [reply front/components/data_source/DataSourceEdition.tsx:36]: Well, this `Modal` is supposed to also handle connection creation, right? so in this case you would get a `DataSourceIntegration`

File: front/components/data_source/DataSourceEdition.tsx [hunk 10]
```diff
@@ -0,0 +1,436 @@
...
+
+  let dataSourceOwner: EditedByUser | null | undefined = null;
+  let isDataSourceOwner: boolean = false;
+  if (isSetup) {
+    dataSourceOwner = dataSourceIntegration.editedByUser;
+    isDataSourceOwner =
+      dataSourceIntegration?.editedByUser?.userId === user.sId;
+  }
+
+  const handleEnableManagedDataSource = async (
```
Comment (tdraier) [inline front/components/data_source/DataSourceEdition.tsx]: Isn't this code only used for creating a connection ?

**Review by tdraier — APPROVED** [review]

**Pull request merged**